File: dollarx/src/main/java/com/github/loyada/dollarx/RelationOperator.scala
package com.github.loyada.dollarx

object RelationOperator extends Enumeration {
  type RelationOperator = Value
  val exactly, orMore, orLess = Value
  def opAsXpathString(relationOperator: RelationOperator) = {
    val sByOp = Map(exactly -> "=", orMore -> ">=", orLess -> "<=")
    sByOp(relationOperator)
  }
  def opAsEnglish(relationOperator: RelationOperator) = {
    val sByOp = Map(exactly -> " ", orMore -> " at least ", orLess -> " at most ")
    sByOp(relationOperator)
  }
}

File: jdollarx/src/main/java/com/github/loyada/jdollarx/RelationOperator.java
package com.github.loyada.jdollarx;

/**
 * Internal implementation.
 */
public enum RelationOperator {
    exactly, orMore, orLess;

    public static String opAsXpathString(RelationOperator op) {
        switch (op) {
            case exactly:
                return "=";
            case orMore:
                return ">=";
            case orLess:
                return "<=";
            default:
                throw new IllegalArgumentException();
        }
    }

    public static String opAsEnglish(RelationOperator op) {
        switch (op) {
            case exactly:
                return " ";
            case orMore:
                return " at least ";
            case orLess:
                return " at most ";
            default:
                throw new IllegalArgumentException();
        }
    }

}
